### PR TITLE
Issue72

### PR DESCRIPTION
**Pull request by xumingming**
implemented the java version of 'testing'

### CODE REVIEW

**Comment by nathanmarz** [comment]
A few comments:
1. Let's make this work with LocalCluster. In the Java interface, you can make it take in a ILocalCluster. You can then add extra methods to LocalCluster not exposed in ILocalCluster that your code can use to extract the cluster map within it.
2. Let's write the TestingTest in Clojure, not Java :) That way it runs with the regular Storm test suite.
3. There's no need to expose the clusterState object in LocalCluster.

**Comment by xumingming** [comment]
By "make this work with LocalCluster", you mean let the method defined in testing4j take a LocalCluster as param? to replace the `backtype.storm.testing.Cluster`?

**Comment by xumingming** [comment]
My intention to write the unit test in java was it might also be a good sample to storm user telling them how to use the api, I will change them into clojure and move the sample code maybe into `storm-lib`

**Comment by nathanmarz** [comment]
Yes, that's right. Replace the usage of backtype.storm.testing.Cluster with LocalCluster in the API.

Also, makes sense regarding converting the test to clojure for unit testing and moving the Java version to an example project.

**Comment by xumingming** [comment]
converted the unit test to clojure and replace the `testing.Cluster` with `ILocalCluster`

File: src/jvm/backtype/storm/utils/Utils.java [hunk 1]
```diff
@@ -342,4 +342,13 @@ public static CuratorFramework newCuratorStarted(Map conf, List<String> servers,
         buffer.get(ret, 0, ret.length);
         return ret;
     }
+    
+    public static List<Object> list(Object... objects) {
```
Comment (nathanmarz) [inline src/jvm/backtype/storm/utils/Utils.java]: This function is unnecessary. Just use the Values class which already does this. "new Values(obj1, obj2, obj3)". Values implements List.

File: src/jvm/backtype/storm/ILocalCluster.java [hunk 2]
```diff
@@ -24,4 +24,6 @@
     StormTopology getTopology(String id);
     ClusterSummary getClusterInfo();
     TopologyInfo getTopologyInfo(String id);
+    Map getState();
+    Object getNimbus();
```
Comment (nathanmarz) [inline src/jvm/backtype/storm/ILocalCluster.java]: I don't think this method is necessary on the interface. The only place that uses it is the Testing#submitLocalTopology method, and that can just be gotten rid of.

File: src/clj/backtype/storm/testing4j.clj [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  ([^Map result ^String componentId ^String streamId]
+     (let [stream-result (.get result componentId)
+           ret (if stream-result
+                 (find-tuples stream-result streamId)
+                 [])]
+       ret))
+  ([^Map result ^String componentId]
+     (-readTuples result componentId Utils/DEFAULT_STREAM_ID)))
+
+(defn -submitLocalTopology [^ILocalCluster cluster ^String topologyName ^Config config ^StormTopology topology]
```
Comment (nathanmarz) [inline src/clj/backtype/storm/testing4j.clj]: This method isn't needed, as users can just use LocalCluster#submitTopology method

**Comment by xumingming** [comment]
Yeah, indeed `submitLocalTopology`, `Utils.list` are not needed, I removed them.

**Comment by nathanmarz** [comment]
Merged! I made a few minor changes with respect to naming/types. I also uncommented some unit test assertions that were commented for some reason.

**Pull request merged**